src/sophys/common/plans/parallel_wrapper.py

```diff
@@ -0,0 +1,44 @@
+from threading import Event, Thread
+
+
+def yield_msg(msg):
+    yield msg
+
+def all_plans_have_finished(plan_has_finished_list):
+    for plan_finished in plan_has_finished_list.values():
+        if not plan_finished.is_set():
+            return True
+    return False
```

Comment:
Maybe this implementation can be simplified using `any`/`all`? For example, to check that all have finished: `all((plan_finished.is_set() for plan_finished in plan_has_finished_list.values()))`